woof-code/rootfs-skeleton/etc/rc.d/rc.shutdown

```diff
@@ -130,10 +136,27 @@ for i in /sys/class/net/* ; do
 	fi
 done
 
-#terminate dbus daemon
-killall dbus-daemon 2>/dev/null
+#=======================================================================
 
-killall udevd > /dev/null 2>&1
+# terminate stuff
+# dbus daemon
+if [ -e /var/run/dbus/dbus.pid ]; then
```

Comment:
dpup still uses /var/run/dbus/pid and still suffers from the issue of D-Bus that refuses to start if it exists, despite https://github.com/puppylinux-woof-CE/woof-CE/commit/76a6e9a660b7e624d0a89406fafebab734c1b227#diff-c75f6042f552eb9e4ab0a07074a889c068ba6f3c9f027b429b075758ae456bbfR23 :(

Reply:
what about a loop?

``` sh
if [ -d /var/run/dbus/ ]; then
	for p in  /var/run/dbus/*.pid 
	do
		DPID=$(cat $p)
		echo killing $DPID
		kill $DPID >/dev/null 2>&1
		rm -f $p
	done
fi
```

Reply:
Maybe `/var/run/dbus/*pid`, without the dot.

I just reverted that commit, because Debian's /etc/init.d/dbus deletes the pidfile if the PID in it doesn't belong to a running process. If the problem persists, a loop like this (or a simple `rm /var/run/dbus/*` in rc.sysinit) can fix the issue.

Reply:
Ok, I'll add the loop but for a failsafe, say power failure, it should be added to rc.sysinit as well.